scripts/src/admin/services.ts

```diff
@@ -405,22 +411,23 @@ export async function getApplicationUserData(params: { app_user_id: string, app_
 }
 
 export async function getOrders(params: any, query: Paging & { status?: OpenOrderStatus, user_id?: string, offer_id?: string }): Promise<string> {
-	const queryBy: { offerId?: string, userId?: string, status?: OpenOrderStatus } = {};
+	const q = await Order.queryBuilder("order");
+
 	if (query.offer_id) {
-		queryBy.offerId = query.offer_id;
-	}
-	if (query.user_id) {
-		queryBy.userId = query.user_id;
+		q.andWhere("offer_id = :offer_id", { offer_id: query.offer_id });
 	}
 	if (query.status) {
-		queryBy.status = query.status;
+		q.andWhere("status = :status", { status: query.status });
 	}
-	const orders = await Order.find({
-		where: queryBy,
-		order: { currentStatusDate: "DESC" },
-		take: take(query),
-		skip: skip(query)
-	});
+	if (query.user_id) {
+		const contexts = await OrderContext.find({ userId: query.user_id });
+		if (contexts.length > 0) {
```

Comment:
`if (contexts.length) {`

Reply:
but maybe do:
`contexts.length ? q.andWhere("id in (:ids)", { ids: contexts.map(c => c.orderId) }) : q.andWhere("id = 'no-such-id' ");`
Feels nicer when it's all on one line. I don't think it's too much.

Reply:
What's `'no-such-id'` btw?

Reply:
If a user has no orders, an empty list should return. `WHERE id IN ()` is an illegal SQL, so I'm just using an operation that will never return. I could of also just run `WHERE 1 = 0`